File: reddit_challenges/20170519-277/277.c
#include <stdio.h>
#include <stdlib.h>

int gcd(int, int);

int main(int argc, char **argv) {
  int num = atoi(argv[1]);
  int den = atoi(argv[2]);
  int res = gcd(num, den);

  printf("%d %d\n", num/res, den/res);

  return 0;
}

int gcd(int a, int b) {
  int i;

  if((a-b) < 0)
    i=a;
  else
    i=b;

  for(i=a; i>1; i--) {
    if(a%i == 0 && b%i == 0)
      return i;
  }
  return 1;
}

File: reddit_challenges/20170519-277/277.py
def gcd(a, b):

	if a-b < 0:
		x = a
	else:
		x = b

	for i in range(x, 2, -1):
		if a%i == 0 and b%i == 0:
			return i

	return 1

a = 1536
b = 78360
c = gcd(a,b)

print("gcd=", c, "num=", a/c, "den=", b/c)
